### PR TITLE
feat(EMI-1718): add toggle component

### PR DESCRIPTION
**Pull request by rquartararo**
[EMI-1718](https://artsyproduct.atlassian.net/browse/EMI-1718)

This PR adds the new Toggle component from the [design system](https://www.figma.com/file/gZNkyqLT8AU3T61tluVJyB/Artsy-3.1-Design-System?type=design&node-id=16243-1514&mode=dev)

@artsy/emerald-devs 

### Demo

https://github.com/artsy/palette/assets/50849237/2dfbcef5-3c6d-45b5-a943-8174491ebaec

### States
Screenshot 2024-03-05 at 9 27 18 AM

[EMI-1718]: https://artsyproduct.atlassian.net/browse/EMI-1718?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by MrSltun — APPROVED** [review]
Looks great! ✨

**Review by dzucconi — COMMENTED** [review]
Looks good to me. But I don't see this in [the design system](https://www.figma.com/file/gZNkyqLT8AU3T61tluVJyB/Artsy-3.1-Design-System?node-id=604%3A1770&mode=dev)? Where is the spec?

**Review by fladson — APPROVED** [review]

File: packages/palette/src/elements/Toggle/Toggle.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  ...rest
+}) => {
+  const isSelectable = !disabled && onSelect !== undefined
+
+  const handleClick = (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+    if (isSelectable && onSelect) {
+      onSelect(!selected)
+    }
+
+    if (onClick !== undefined) {
```
Comment (damassi) [inline packages/palette/src/elements/Toggle/Toggle.tsx:35]: No need for anything more than this: 
```tsx
if (onClick) {
  onClick()
}
```

**Review by damassi — APPROVED** [review]
Nice 👍

File: packages/palette/src/elements/Toggle/Toggle.tsx (hunk 1 above)
Comment (dzucconi) [reply packages/palette/src/elements/Toggle/Toggle.tsx:35]: `onClick?.()`

**Comment by artsyit** [comment]
:rocket: PR was released in [`@artsy/palette-charts@37.3.0`](https://github.com/artsy/palette/releases/tag/%40artsy/palette-charts%4037.3.0), [`@artsy/palette@38.3.0`](https://github.com/artsy/palette/releases/tag/%40artsy/palette%4038.3.0) :rocket:

File: packages/palette/src/elements/Toggle/Toggle.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  }
+
+  return (
+    <Container
+      display="flex"
+      alignItems="center"
+      onClick={handleClick}
+      tabIndex={disabled ? -1 : 0}
+      onKeyPress={handleKeyPress}
+      role="toggle"
```
Comment (dzucconi) [inline packages/palette/src/elements/Toggle/Toggle.tsx:54]: I believe this should be `role="switch"` https://www.w3.org/WAI/ARIA/apg/patterns/switch/

**Pull request merged**